### ISSUE TITLE
Can BatchDecodeWithPaddedKVCache be used in cascade inference?

### ISSUE THREAD

**Issue by joey12300** [flashinfer-ai/flashinfer#250]
Currently the implementation of cascade inference is two level, composed of SinglePrefillWithKVCache(SingleDecodeWithKVCache) and 
 BatchPrefillWithPagedKVCacheWrapper(BatchDecodeWithPagedKVCacheWrapper).  I wonder if the batch prefill or decode attention use **without paged attention**?

**Comment by yzh119** [comment]
Yes I do think so, there are some test cases https://github.com/flashinfer-ai/flashinfer/blob/7e9cc7ff42ca283c317061a877305d09a395fad2/python/tests/test_shared_prefix_kernels.py#L33-L53

**Comment by yzh119** [comment]
Padded KV-Cache was deprecated in https://github.com/flashinfer-ai/flashinfer/releases/tag/v0.0.7.

For cascade inference, it's encouraged to use the new [MultiLevelCascadeAttentionWrapper](https://docs.flashinfer.ai/api/python/cascade.html#flashinfer.cascade.MultiLevelCascadeAttentionWrapper) API introduced in https://github.com/flashinfer-ai/flashinfer/releases/tag/v0.1.6
